client/conn.go

```diff
@@ -108,19 +116,21 @@ func ConnectWithDialer(ctx context.Context, network string, addr string, user st
 	c.password = password
 	c.db = dbName
 	c.proto = network
-	c.Conn = packet.NewConn(conn)
 
 	// use default charset here, utf-8
 	c.charset = DEFAULT_CHARSET
 
 	// Apply configuration functions.
-	for i := range options {
-		options[i](c)
+	for _, option := range options {
```

Comment:
I just found we forget to close the `conn` in line 110 when error happens 😂 Do you think we can fix it in this PR or a separate one?

Reply:
added `Close()` to 3 additional spots in this function.